badge_server/badge_server.py

```diff
@@ -279,6 +307,75 @@ def self_compatibility_badge_target():
     return str(self_comp_res)
 
 
+@app.route('/self_dependency_badge/image')
+def self_dependency_badge_image():
+    """Badge showing whether a package is has outdated dependencies."""
+
+    package_name = flask.request.args.get('package')
+
+    res = {
+        'status': 'CALCULATING',
+        'details': None,
+    }
+
+    def run_check():
+        # First see if this package is already stored in BigQuery.
+        outdated = highlighter.check_package(package_name)
+
+        max_level = priority_level.UP_TO_DATE
+        details = []
+        for dep in outdated:
+            level = dep.priority.level
+            if level.value > max_level.value:
```

Comment:
Why having this? The details could just be `None` if the version is up to date.

Reply:
this snippet does 2 things:
1. track the highest outdated dependency (to be displayed on the badge)
2. append detail string for every outdated dependency to the list

if the list is empty, that means no outdated dependencies and the package is up to date and details is `EMPTY_DETAILS`

if the list is not empty, the list of strings are joined together to produce one string and provides info on all dependencies that are outdated

Reply:
Sounds good, thanks for the explanation! And please also reply `done` or any comment you prefer to the rest of the review comments in future PRs, which makes it easier to track which were resolved. Thanks!

Reply:
ok! thanks for the review!